android/src/main/java/xyz/luan/audioplayers/WrappedSoundPool.java

```
@@ -0,0 +1,246 @@
+    private Integer streamId;
+
+    private boolean playing = false;
+
+    private boolean paused = false;
+
+    private boolean looping = false;
+
+    private boolean loading = false;
+
+    WrappedSoundPool(AudioplayersPlugin ref, String playerId) {
+        this.ref = ref;
+        this.playerId = playerId;
+    }
+
+    @Override
+    String getPlayerId() {
+        return null;
+    }
+
+    @Override
+    void play() {
+        if (this.playing) {
+            return;
+        }
+        if (!this.loading) {
+            start();
+        }
+        this.playing = true;
+    }
+
+    @Override
+    void stop() {
+        if (this.playing) {
+            soundPool.stop(this.streamId);
+            this.playing = false;
+        }
+        this.paused = false;
+    }
+
+    @Override
+    void release() {
+        this.stop();
+        soundPool.unload(this.soundId);
+    }
+
+    @Override
+    void pause() {
+        if (this.playing) {
+            soundPool.pause(this.streamId);
+            this.playing = false;
+            this.paused = true;
+        }
+    }
+
+    @Override
+    void setUrl(final String url) {
+        if (this.url != null && this.url.equals(url)) {
+            return;
+        }
+        if (this.soundId != null) {
+            soundPool.unload(this.soundId);
+        } else {
+            soundPool.setOnLoadCompleteListener(this);
+        }
+        this.url = url;
+        this.loading = true;
+
+        final WrappedSoundPool self = this;
+
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                self.soundId = soundPool.load(getAudioPath(url), 1);
+            }
+        }).start();
+    }
+
+    @Override
+    void setVolume(double volume) {
+        this.volume = (float) volume;
+        if (this.playing) {
+            soundPool.setVolume(this.streamId, this.volume, this.volume);
+        }
+    }
+
+    @Override
+    void configAttributes(boolean respectSilence) {
+
```

wouldn't be better to throw errors when using the unavailable ops, instead of failing silently? just a question, though, don't know the implications of doing that